_episodes/11-dask-configuration.md

```diff
@@ -0,0 +1,390 @@
+```
+
+Open the Dashboard link in a browser to see the Dask Dashboard website.
+When the recipe has finished running, the Dashboard website will stop working.
+The top left panel shows the memory use of each of the workers, the panel on the
+right shows one row for each thread that is doing work, and the panel at the
+bottom shows the progress of all work that the scheduler currently has been
+asked to do.
+
+> ## Explore what happens if workers do not have enough memory
+>
+> Reduce the amount of memory that the workers are allowed to use to 2GiB and
+> run the recipe again. Watch what happens.
+>
+>> ## Solution
+>>
+>> We use `memory_limit` entry in the `~/.esmvaltool/dask.yml` file to set the
+>> amount of memory allowed to 2GiB:
+>>```yaml
+>> cluster:
+>>    type: distributed.LocalCluster
+>>    n_workers: 1
+>>    threads_per_worker: 2
+>>    memory_limit: 2GiB
+>>```
+>> Note that the bars representing the memory use turn
+>> orange as the worker reaches the maximum amount of memory it is
+>> allowed to use and it starts 'spilling' (writing data temporarily) to disk.
+>> The red blocks in the top right panel represent time spent reading/writing
+>> to disk. While 2 GiB per worker may be enough in other cases, it is
+>> apparently not enough for this recipe.
+>>
+> {: .solution}
+{: .challenge}
+
+
+> ## Tune the configuration to your own computer
+>
+> Look at how much memory you have available on your machine (e.g. by running
+> the command ``grep MemTotal /proc/meminfo`` on Linux), set the
+> ``memory_limit`` back to 4 GiB per worker and increase the number of Dask
+> workers so they use total amount available minus a few gigabytes for your
+> other work. Run the recipe again and notice that it completed faster.
+>
+>> ## Solution
+>>
+>> For example, if your computer has 16 GiB of memory and you do not have too
+>> many other programs running, it can use 12 GiB of memory for Dask workers,
+>> so you can start 3 workers with 4 GiB of memory each.
+>>
+>> Use the `num_workers` entry in the `~/.esmvaltool/dask.yml` file to set the
+>> number of workers to 3:
+>>```yaml
+>> cluster:
+>>    type: distributed.LocalCluster
+>>    n_workers: 3
+>>    threads_per_worker: 2
+>>    memory_limit: 4GiB
+>>```
+>> and run the recipe again with the command
+>> ``esmvaltool run recipe_easy_ipcc_short.yml``.
+>> The time it took to run the recipe is printed to the screen.
+>>
+> {: .solution}
+{: .challenge}
+
+## Using an existing Dask Distributed cluster
+
+In some cases, it can be useful to start the Dask Distributed cluster before
+running the ``esmvaltool`` command. For example, if you would like to keep the
+Dashboard available for further investigation after the recipe completes
+running, or if you are working from a Jupyter notebook environment, see
+[dask-labextension](https://github.com/dask/dask-labextension) and
+[dask_jobqueue interactive use](https://jobqueue.dask.org/
+en/latest/interactive.html)
+for more information.
+
```

Comment:
Can the example below be part of a Pro-tip? We put some more advanced material as tips like that in other episodes.

Reply:
Setting up the Dask Distributed cluster separately seems very useful for educational purposes, as it allows the training participants to play around with the dashboard at their leisure. Do you think the Python code is too intimidating?

Reply:
I think it might be but it is also useful, so perhaps good for a Pro-tip?

Reply:
I made the section a Pro-tip, does it look better now?